src/access/RequestPath.h

```diff
@@ -19,15 +19,29 @@
 #pragma once
 
 #include <lib/core/DataModelTypes.h>
+#include <optional>
 
 namespace chip {
 namespace Access {
 
+enum class RequestType : uint8_t
+{
+    kRequestTypeUnknown,
+    kReadRequest,
+    kWriteRequest,
+    kInvokeRequest,
+    kSubscribeEventRequest
+};
+
 struct RequestPath
 {
     // NOTE: eventually this will likely also contain node, for proxying
-    ClusterId cluster   = 0;
-    EndpointId endpoint = 0;
+    ClusterId cluster       = 0;
+    EndpointId endpoint     = 0;
+    RequestType requestType = RequestType::kRequestTypeUnknown;
+
+    //entityId represents an attribute, command, or event ID, which is determined by the requestType. Wildcard if omitted.
+    std::optional<uint32_t> entityId;
```

Comment:
So wildcarding is used here only for "reading or subscribing to events" and "it's a wildcard path", right?

But then how is this meant to be implemented, exactly?  We really do need to know what events are involved to do the relevant checks correctly for ARL, no?

Reply:
A wildcard here would only make sense when reading events or attributes, or subscribing to them, yes.

If a request is made using a wildcard, restriction checks would fail if there is any intersection between that and what is restricted.  Meaning, a single attribute/event restriction entry would  block a wildcard request if that wildcard would have included the entry.

Make sense?  Or am I missing something?

Reply:
That's not how wildcards are supposed to work with access check failures.  The way it's _supposed_ to work is that you just exclude anything you don't have access to from the set of things you actually get reports on.  If that results in an empty set, the whole subscription is denied.

But doing whole-node wildcard subscriptions is the _normal_ way controllers operate.  If you disallow those if there is "any intersection", you would break every controller out there.

Reply:
Got it.  Wildcard subscriptions will succeed.  The check will be on the event read and at time of event delivery.